lib/sensors/mux.cpp

```diff
@@ -53,8 +55,15 @@ core::Result Mux<T, N>::closeAllChannels()
 template<typename T, std::uint8_t N>
 std::optional<std::array<T, N>> Mux<T, N>::readAllChannels()
 {
-  std::array<T, N> mux_data;
+  // Zero-initialize the array
+  std::array<T, N> mux_data{};
   for (std::uint8_t i = 0; i < N; ++i) {
+    if (num_unusable_sensors > max_num_unusable_sensors_) {
```

Comment:
This should move to the end; what if the last sensor is unusable and reaches the threshold?